lxd/certificates.go

```diff
@@ -434,13 +434,20 @@ func certificatesPost(d *Daemon, r *http.Request) response.Response {
 			return response.Forbidden(nil)
 		}
 
-		// A password is required for non-admin users.
-		if req.Password == "" {
+		// A password/token is required for non-admin users.
+		if req.Password == "" && req.TrustToken == "" {
 			return response.Forbidden(nil)
 		}
 
-		// Check if cluster member join token supplied as password.
-		joinToken, err := shared.JoinTokenDecode(req.Password)
+		var joinTokenEncoded string
+		if req.Password != "" {
+			joinTokenEncoded = req.Password
+		} else if req.TrustToken != "" {
```

Comment:
Should we prefer TrustToken over Password fields if both are provided (or fail if both are provided)?

Reply:
Failing seems more reasonable as specifying both might indicate LXD is using both methods (one after another) but it is only using one of them. I'll add a follow up PR.